src/Bicep.Core/Emit/ForSyntaxValidatorVisitor.cs

```diff
@@ -135,6 +135,13 @@ public override void VisitOutputDeclarationSyntax(OutputDeclarationSyntax syntax
             this.activeLoopCapableTopLevelDeclaration = null;
         }
 
+        public override void VisitAssertDeclarationSyntax(AssertDeclarationSyntax syntax)
```

Comment:
>VisitAssertDeclarationSyntax [](http://example.com/codeflow?start=29&length=28)

This visitor validates whether a for-expression is allowed. Since asserts expect an expression of bool type, the value can't be an expression like `[for x in myList]`.

I think you can safely remove the change, but will need to verify that the for-expression is still blocked in assert values.